### PR TITLE
feat(dotcom-shell): added a story featuring horizontal-toc

### PR DESCRIPTION
**Pull request by RaphaelAmadeu**
### Description

I've added a story under Dotcom-shell that features the horizontal top and leadspace.

### Related issue
#5696

### CODE REVIEW

**Comment by shixiedesign** [comment]
Not sure if this is ready for review yet, but here's what I noticed: 
1. Quite a big space between lead space and ToC. Should be 0 gap there

image

2. When clicking on ToC, section scrolls too far and disappears a bit under the ToC. Should scroll less so all of the section's header is visible, and 32px from ToC

3. L0 should unstick & scroll away when ToC becomes sticky. 

image

**Comment by RaphaelAmadeu** [comment]
> Not sure if this is ready for review yet, but here's what I noticed:
> 
> 1. Quite a big space between lead space and ToC. Should be 0 gap there
> 
> image
> 
> 1. When clicking on ToC, section scrolls too far and disappears a bit under the ToC. Should scroll less so all of the section's header is visible, and 32px from ToC
> 2. L0 should unstick & scroll away when ToC becomes sticky.
> 
> image

Hey Shixie, currently this story is being created to test how both headspace and horizontal toc are working together, so these points should be addressed in separate prs

**Comment by photodow** [comment]
> Hey Shixie, currently this story is being created to test how both headspace and horizontal toc are working together, so these points should be addressed in separate prs

I'm all for iterative approaches, but I'm wondering if we should wait to merge the story in for the public to see until some of these major bugs are fixed?

**Comment by shixiedesign** [comment]
Do you recommend a separate PR because the bug is happening outside this `Dotcom shell` component? Is it happening in ToC? We just filed a lot bugs in ToC and I don't remember seeing this. If this bug only surfaces in Dotcom shell and no where else, I think we should still fix in the same PR, just to make sure new things are not shipped broken since the start.

**Comment by shixiedesign** [comment]
Is it related to this story? https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5569

**Comment by shixiedesign** [comment]
I gotcha, okay then let's quickly get to the ToC and lead space issue fixes. Are these fixes within the scope of this PR? 

- Body content should be sitting on the grid, currently 16px too far to the right 
image

- Remove knob for Platform. We do not allow this option to our users. Even Cloud has a specific design that's not this. When we get to clean up the Dotcom shell story, we will look into removing it all around.
image

- Rename sub-story to "With ToC horizontal" 

- Knob `selected menu item` should be a dropdown with existing options in L0, not a text-input 
image

- Knobs should not allow tabs. Add the knobs in "FooterComposite" into the same tab and remove tabs. In general avoiding tabs help with discoverability. 
image

### So in summary, the Dotcom shell - story should look like this: 

```
Dotcom shell
|--- With ToC horizontal 
```

### Knobs in the story are: 

| Name | Type | Default value | 
| --- | --- | --- |
| Show profile in masthead | Checkbox | Checked | 
| Show search in masthead | Checkbox | Checked |
| Search placeholder | Text input | Search all of IBM | 
| Selected menu item | Dropdown | Services & Consulting |
| User authentication | Dropdown | unauthenticated |

Thanks!

**Comment by shixiedesign** [comment]
Hi Raph, this is great. The only thing missed is this! Sorry to be nitpick. If this is not possible, we can also address later along with platform.
- [ ] Knob selected menu item **should be a dropdown** with existing options in L0, not a text-input
image
image

**Comment by jeffchew** [comment]
Hey @RaphaelAmadeu just checking on this, this PR is a couple of weeks old now so wanted to see if we can get this moved in? Also looks to be some ci-check failures if you can take a look!

**Review by shixiedesign — APPROVED** [review]
There was another bug I picked on earlier but we can address later in another round of storybook refinement. Better not let this go stale! @jeffchew @RaphaelAmadeu so if dev review approves, this is good to merge!

**Comment by jeffchew** [comment]
Hey @RaphaelAmadeu , can you look to see why the web components ci-check is failing?

**Comment by jeffchew** [comment]
@RaphaelAmadeu the `language only` stories seemed to have been affected by this, they are loading the locale modal button and not the language only selector. can you check?

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 1]
```diff
@@ -691,26 +837,23 @@ export default {
...
         platform: select('Platform (platform)', { none: null, platform: platformData.name }, null, groupId),
-        hasProfile: boolean('show the profile functionality (profile)', true, groupId),
-        hasSearch: boolean('show the search functionality (search)', true, groupId),
-        searchPlaceholder: textNullable('search placeholder (searchPlaceholder)', inPercy() ? ' ' : 'Search all of IBM', groupId),
-        selectedMenuItem: textNullable('selected menu item (selected-menu-item)', 'Services & Consulting', groupId),
-        userStatus: select('The user authenticated status (user-status)', userStatuses, userStatuses.unauthenticated, groupId),
-      }),
-      FooterComposite: ({ groupId }) => ({
-        disableLocaleButton: boolean('hide the locale button (disable-locale-button)', false, groupId),
+        hasProfile: boolean('Show profile in masthead (profile)', true, groupId),
```
Comment (kennylam) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:842]: `platform` for this story can be removed, per [Shixie's comment](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5565#issuecomment-811201753).

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 2]
```diff
@@ -691,26 +837,23 @@ export default {
...
+        hasSearch: boolean('Show search in masthead (search)', true, groupId),
+        searchPlaceholder: textNullable('Search placeholder (searchPlaceholder)', 'Search all of IBM', groupId),
+        selectedMenuItem: textNullable('Selected menu item (selected-menu-item)', 'Services & Consulting', groupId),
```
Comment (kennylam) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:845]: This should be a `select()`, per [Shixie's comment](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5565#issuecomment-811201753) – can just create a simple array with the existing menu items, maybe?

File: packages/web-components/src/components/dotcom-shell/__stories__/data/content.ts [hunk 3]
```diff
@@ -254,6 +258,115 @@ const StoryContent = () => html`
       </dds-table-of-contents>
     </div>
   </main>
-`;
+`
```
Comment (kennylam) [inline packages/web-components/src/components/dotcom-shell/__stories__/data/content.ts:261]: Can this entire `html` section get some format-love please?

**Review by kennylam — COMMENTED** [review]
Thanks for this @RaphaelAmadeu! Just a few small comments from me.

**Comment by jeffchew** [comment]
Hey @RaphaelAmadeu ! Just checking on status of this, thanks!

**Review by kennylam — APPROVED** [review]
Looks good thanks @RaphaelAmadeu!
